linkerd/http/prom/src/body_data/metrics.rs

```diff
@@ -0,0 +1,94 @@
+//! Prometheus counters for request and response bodies.
+
+use linkerd_metrics::prom::{self, Counter, Family, Registry};
+
+/// Counters for response body frames.
+#[derive(Clone, Debug)]
+pub struct ResponseBodyFamilies<L> {
+    /// Counts the number of response body frames.
+    resp_body_frames_total: Family<L, Counter>,
+    /// Counts the total number of bytes in response body frames.
+    resp_body_frames_bytes: Family<L, Counter>,
+}
+
+/// Counters to instrument a request or response body.
+#[derive(Clone, Debug, Default)]
+pub struct BodyDataMetrics {
+    /// Counts the number of request body frames.
+    pub frames_total: Counter,
+    /// Counts the total number of bytes in request body frames.
+    pub frames_bytes: Counter,
+}
+
+// === impl ResponseBodyFamilies ===
+
+impl<L> Default for ResponseBodyFamilies<L>
+where
+    L: Clone + std::hash::Hash + Eq,
+{
+    fn default() -> Self {
+        Self {
+            resp_body_frames_total: Default::default(),
+            resp_body_frames_bytes: Default::default(),
+        }
+    }
+}
+
+impl<L> ResponseBodyFamilies<L>
+where
+    L: prom::encoding::EncodeLabelSet
+        + std::fmt::Debug
+        + std::hash::Hash
+        + Eq
+        + Clone
+        + Send
+        + Sync
+        + 'static,
+{
+    const RESP_BODY_FRAMES_TOTAL_NAME: &'static str = "resp_body_frames_total";
+    const RESP_BODY_FRAMES_TOTAL_HELP: &'static str =
+        "Counts the number of frames in response bodies.";
+
+    const RESP_BODY_FRAMES_BYTES_NAME: &'static str = "resp_body_frames_bytes";
+    const RESP_BODY_FRAMES_BYTES_HELP: &'static str =
+        "Counts the total number of bytes in response bodies.";
```

Comment:
the name and help strings for the new metrics added here.

Reply:
i'm lightly in favor of them, because they explicitly declare to readers what the name and help strings are for the metrics emitted by this middleware. while they are not _mechanically_ required, i find this pattern to be friendly and helpful for readers trying to reason about the metrics provided by a piece of middleware, particularly if they are not Rust experts themselves.

while i have a preference for `const`s like this however, i agree with the desire for a convergent style if that outweighs the benefit, and am happy to remove them. what do you think?